client/files_test.go

```diff
@@ -0,0 +1,280 @@
+      "last-modified": "2022-08-22T12:42:49Z",
+      "user-id": 0,
+      "user": "root",
+      "group-id": 0,
+      "group": "root"
+	},
+	{
+      "path": "/dbus.sock",
+      "name": "dbus.sock",
+      "type": "named-pipe",
+      "permissions": "700",
+      "last-modified": "2022-08-22T12:42:49Z",
+      "user-id": 0,
+      "user": "root",
+      "group-id": 0,
+      "group": "root"
+	},
+	{
+      "path": "/tty0",
+      "name": "tty0",
+      "type": "device",
+      "permissions": "700",
+      "last-modified": "2022-08-22T12:42:49Z",
+      "user-id": 0,
+      "user": "root",
+      "group-id": 0,
+      "group": "root"
+	},
+	{
+      "path": "/irreg",
+      "name": "irreg",
+      "type": "sfdeljknesv",
+      "permissions": "700",
+      "last-modified": "2022-08-22T12:42:49Z",
+      "user-id": 0,
+      "user": "root",
+      "group-id": 0,
+      "group": "root"
+	}
+  ]
+}`
+	result, err := cs.cli.ListFiles(&client.ListFilesOptions{
+		Path: "/",
+	})
+	c.Assert(err, IsNil)
+	c.Assert(result, HasLen, 10)
+}
+
+func (cs *clientSuite) TestListDirectoryItself(c *C) {
+	cs.rsp = `{
+  "type": "sync",
+  "status-code": 200,
+  "status": "OK",
+  "result": [
+    {
+      "path": "/bin",
+      "name": "bin",
+      "type": "symlink",
+      "permissions": "777",
+      "last-modified": "2022-04-21T03:02:51Z",
+      "user-id": 0,
+      "user": "root",
+      "group-id": 0,
+      "group": "root"
+    }
+  ]
+}`
+	result, err := cs.cli.ListFiles(&client.ListFilesOptions{
+		Path:   "/bin",
+		Itself: true,
+	})
+	c.Assert(err, IsNil)
+	c.Assert(result, HasLen, 1)
+
+	c.Check(result[0].Name(), Equals, "bin")
+	c.Check(result[0].Size(), Equals, int64(0))
+	c.Check(result[0].Mode(), Equals, 0o777|os.ModeSymlink)
+	c.Check(result[0].ModTime(), DeepEquals, time.Date(2022, 4, 21, 3, 2, 51, 0, time.UTC))
+	c.Check(result[0].Path(), Equals, "/bin")
+	c.Check(result[0].UserID(), Equals, 0)
+	c.Check(result[0].GroupID(), Equals, 0)
+	c.Check(result[0].User(), Equals, "root")
+	c.Check(result[0].Group(), Equals, "root")
+	c.Check(result[0].IsDir(), Equals, false)
+	c.Check(result[0].Sys(), Equals, nil)
+}
+
+func (cs *clientSuite) TestListFilesFails(c *C) {
+	cs.rsp = `{
+  "type": "error",
+  "result": {
+    "message": "could not foo"
+  }
+}`
+	_, err := cs.cli.ListFiles(&client.ListFilesOptions{
+		Path:   "/",
+		Itself: true,
+	})
+	c.Assert(err, ErrorMatches, "could not foo")
+}
+
+func (cs *clientSuite) TestCalculateFileModeFails(c *C) {
+	cs.rsp = `{
+  "type": "sync",
+  "status-code": 200,
+  "status": "OK",
+  "result": [
+    {
+      "path": "/irreg",
+      "name": "irreg",
+      "type": "sfdeljknesv",
+      "permissions": "deadbeef",
+      "last-modified": "2022-08-22T12:42:49Z",
+      "user-id": 0,
+      "user": "root",
+      "group-id": 0,
+      "group": "root"
+	}
+  ]
+}`
+	_, err := cs.cli.ListFiles(&client.ListFilesOptions{
+		Path: "/irreg",
+	})
+	c.Assert(err, ErrorMatches, "*invalid syntax*")
```

Comment:
I'm not sure I understand how this matches, as `*` isn't a valid regular expression, and `ErrorMatches` takes a regex, right? Wouldn't it be `.*`?

Reply:
Mere luck ;)

<img width="354" alt="Captura de pantalla 2022-08-24 a las 12 19 15" src="https://user-images.githubusercontent.com/55109673/186394281-10eb404e-3689-4562-8991-e2c8fd454b97.png">